roles/openshift_health_checker/library/etcdstats.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/python
+# vim: expandtab:tabstop=4:shiftwidth=4
+'''
+Ansible module that determines if the size of paths in an etcd cluster exceed a given limit.
+'''
+
+try:
+    import etcd
+except ImportError:
+    # Allows for test mocking
+    from collections import namedtuple
+    EtcdMock = namedtuple("etcd", ["EtcdKeyNotFound"])
+    etcd = EtcdMock(KeyError)
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+# pylint: disable=too-many-arguments
+def check_etcd_key_size(client, key, total_size=0, size_limit=0, depth=0, depth_limit=1000, visited=None):
+    """Recursively checks that total size of an etcd path starting at a given key does not exceed a given limit"""
+    if visited is None:
+        visited = set()
+
+    if key in visited:
+        return 0, False
+
+    visited.add(key)
+
+    try:
+        result = client.read(key, recursive=False)
+    except etcd.EtcdKeyNotFound:
+        return 0, False
+
+    size = 0
+    limit_exceeded = False
+
+    for node in result.leaves:
+        if depth >= depth_limit:
+            raise Exception("Maximum recursive stack depth ({}) exceeded.".format(depth_limit))
+
+        if size_limit and total_size + size > size_limit:
+            return size, True
+
+        if not node.dir:
+            size += len(node.value)
+            continue
+
+        key_size, limit_exceeded = check_etcd_key_size(client, node.key,
+                                                       total_size + size,
+                                                       size_limit,
+                                                       depth + 1,
+                                                       depth_limit, visited)
+        size += key_size
+
+    return size, limit_exceeded or (total_size + size > size_limit)
+
+
+def main():  # pylint: disable=missing-docstring,too-many-branches
+    module = AnsibleModule(
+        argument_spec=dict(
+            size_limit_bytes=dict(type="int", default=0),
+            paths=dict(type="list", default=["/openshift.io/images"]),
+            host=dict(type="str", default="127.0.0.1"),
+            port=dict(type="int", default=4001),
+            protocol=dict(type="str", default="http"),
+            version_prefix=dict(type="str", default=""),
+            allow_redirect=dict(type="bool", default=False),
+            cert=dict(type="dict", default=""),
+            ca_cert=dict(type="str", default=None),
+        ),
+        supports_check_mode=True
+    )
+
+    module.params["cert"] = (
+        module.params["cert"]["cert"],
+        module.params["cert"]["key"],
+    )
+
+    size_limit = module.params["size_limit_bytes"]
+    module.params.pop("size_limit_bytes", None)
```

Comment:
Equivalent to:

    size_limit = module.params.pop("size_limit_bytes")
￼
I'd like all these transform-module-params lines to be in one visual block...